M2/Macaulay2/e/ringelem.hpp

```diff
@@ -94,21 +91,56 @@ union ring_elem
   const schur_poly* get_schur_poly() const { return schur_poly_val; }
 };
 
+/* Implements a linked list of ring monomials along with coefficients */
 struct Nterm
 {
   Nterm *next;
   ring_elem coeff;
+  // TODO: should this have type monomial?
   int monom[1];
```

Comment:
Not sure if I fully understand; isn't `int[1]` also a pointer?

Reply:
t.monom is a pointer, but the actual ints are in the Nterm.  One old-school common trick, in this instance is: An Nterm element in memory is [pointer, ring_elem, int, int, int, int]. (Or some number of int's, depending how the struct is made on the heap).  These days, we still have some major uses of this kind of technique (which removes an extra layer of indirection), but once we rewrite these as not linked list elements anymore, it will not be necessary, but that is for a future refactoring.

Reply:
For now, I'll resolve this conversation, but we can discuss it together later.